### DISCUSSION TITLE
Unmaintained?

### DISCUSSION THREAD

**Discussion by Malix-Labs** [crxjs/chrome-extension-tools#872] [Q&A] [answered]
Hello folks, is this project still maintained?

The docs tells that
> CRXJS support for Vite 3 is in beta
	![image](https://github.com/crxjs/chrome-extension-tools/assets/76160668/7b50e1b8-e2e7-47ac-b866-9285acb63f78)
	https://crxjs.dev/vite-plugin/getting-started/vanilla-js/create-project#create-a-project

Altough [vite 3 was released July 23, 2022](https://github.com/vitejs/vite/releases/tag/v3.0.0), and now it is at [v5.1.4](https://github.com/vitejs/vite/releases/tag/v5.1.4)

Also, [CRXJS latest release is Aug 11, 2022](https://github.com/crxjs/chrome-extension-tools/releases/tag/@crxjs/vite-plugin@1.0.14) and [vite-plugin v2 is in beta since a year and a half](https://github.com/crxjs/chrome-extension-tools/releases/tag/%40crxjs%2Fvite-plugin%402.0.0-beta.1)

**Comment by Toumash** [comment]
Yes it is - the beta version works really well for me with vite 3, 4 and 5.
The question i have currently is - when are we going to be out of beta?

For HMR: im not sure if there are issues with vite 4 but for me it didnt work because of multiple exports - its like that by design. To prevent it use eslint rule https://github.com/ArnaudBarre/eslint-plugin-react-refresh and it should work flawlessy

**Reply by Toumash** [reply]
update: vite 6 also works

**Reply by Malix-Labs** [reply]
I assessed that and yes, it works on vite 6

However, there still was no release while WXT improved much since I opened the post : https://wxt.dev/guide/development.html

I honestly liked CrxJS while it was worked on, and apparently, I'm not the only one
[![image](https://github.com/user-attachments/assets/228e1d17-59ef-4364-9cf9-415bd6510245)](https://github.com/crxjs/chrome-extension-tools/discussions?discussions_q=is%3Aopen+sort%3Atop)

**Comment by Unit2795** [comment]
Works for me just fine with Vite 5, React, and Typescript. Though I am getting a warning when I manage my dependencies. I am using PNPM 9.1.0. Interested to see if this library gets any updates. If anyone ends up finding a newer alternative, keep us all posted. Thank you!

```
 WARN  Issues with peer dependencies found
.
└─┬ @crxjs/vite-plugin 1.0.14
  └── ✕ unmet peer vite@^2.9.0: found 5.2.12
```

![2024-06-03_01-13-02](https://github.com/crxjs/chrome-extension-tools/assets/25918877/af223898-5dc9-4956-b03f-335e0dec1c67)

**Reply by Malix-Labs** [reply]
> If anyone ends up finding a newer alternative, keep us all posted

The two main alternatives are WXT and Plasmo
WXT being better than Plasmo

**Reply by Unit2795** [reply]
> > If anyone ends up finding a newer alternative, keep us all posted
> 
> The two main alternatives are WXT and Plasmo I did go with WXT

Wow, these are fantastic. Thanks a million for sharing! 👍

**Reply by Toumash** [reply]
WXT [does NOT support HMR ](https://github.com/wxt-dev/wxt/issues/357) as is does not have a major version yet. Its a big no-no for me.
Plasmo has a strange implicit way of defining entrypoints but from the github issues it looks like it works (havent tried it yet)
@Unit2795 you have outdated @crxjs/vite-plugin - install the beta release (i dont consider it a beta - i have it for a year on my production) `npm i -D @crxjs/vite-plugin@beta`

**Reply by Malix-Labs** [reply]
> WXT [does NOT support HMR ](https://github.com/wxt-dev/wxt/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)as is does not have a major version yet. Its a big no-no for me.

Yes it does

https://wxt.dev/guide/development.html#:~:text=separately%20when%20possible.-,HMR,-Reloaded%20individually

A big no-no for me is no release in 2y ngl

**Comment by cpakken** [comment]
I wonder how @jacksteamdev is doing? He really made a very useful vite plugin and I think it's still the best way to build chrome extensions to this day even though it's currently unmaintained. I haven't had any major issues with the beta versions. Getting HMR working even with injected content scripts works well. Service workers, content scripts refreshes correctly on save, vite builds bundles assets nicely. What more can you ask for? 

Though I wish for some sort of status update from @jacksteamdev on the future of this plugin.

**Reply by Toumash** [reply]
@AmySteam maybe you have some info about the projects vision?

**Comment by salmin89** [comment]
@Malix-Labs did you end up switching to WXT? any drawbacks?

**Reply by Malix-Labs** [reply]
I did, yes ; but it is not my main project

The [WXT compare](https://wxt.dev/guide/resources/compare.html) is pretty descriptive of the trade-offs

WXT is the current best way to build extensions for sure, no doubt

**Reply by marijnbent** [reply]
https://wxt.dev/guide/resources/migrate.html#crxjs

**Comment by Malix-Labs** [comment, marked as answer]
[CRXJS Will be achieved _(unless someone step up and maintain themself)_](https://github.com/crxjs/chrome-extension-tools/discussions/974)
